src/effects/Wahwah.cpp

```diff
@@ -345,107 +471,138 @@ size_t EffectWahwah::InstanceProcess(EffectSettings &settings,
    return blockLen;
 }
 
-void EffectWahwah::OnFreqSlider(wxCommandEvent & evt)
+void EffectWahwah::Validator::OnFreqSlider(wxCommandEvent & evt)
 {
-   mFreq = (double) evt.GetInt() / Freq.scale;
+   auto& ms = mSettings;
+
+   ms.mFreq = (double) evt.GetInt() / Freq.scale;
    mFreqT->GetValidator()->TransferToWindow();
-   EnableApply(mUIParent->Validate());
+
+   // TODO code this into a single method of class Effect, call it e.g. "EnableApplyFromValidate"
+   mActualEffect.EnableApply(mActualEffect.GetUIParent()->Validate());
```

Comment:
This is incomplete.  Every one of the `::On` functions that handles an event and changes mSettings needs to send that change into the message buffers.  So add a call to ValidateUI at the end of each.

In future, I hope Kit and I will merge a lot of work that will simplify the defintion of effect dialogs and make these event handler functions unnecessary: instead PopulateOrExchange can encode a complete and short description of how values are copied and how one control can be updated to be consistent with another (as with these slider and text pairs).  But, that's future.

Reply:
Ok I missed that. Interestingly, the effect works correctly (applying the right amounts and remembering them) in destructive mode even without that final call to ValidateUI(), but I guess the call would be needed when using it in realtime mode?